runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -279,8 +282,7 @@ boolean atMaxWatermark(DataflowPipelineJob job, JobMetrics metrics) {
   JobMetrics getJobMetrics(DataflowPipelineJob job) {
     JobMetrics metrics = null;
     try {
-      metrics = options.getDataflowClient().projects().jobs()
-          .getMetrics(job.getProjectId(), job.getJobId()).execute();
+      metrics = dataflowClient.getJobMetrics(job.getJobId());
```

Comment:
Would it make sense to allow construction of a dataflowClient that was bound to a job? Seems like the most common case...